src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -374,21 +381,27 @@ public override Task<bool> IsSignedAsync(CancellationToken token)
 
             ThrowIfZipReadStreamIsNull();
 
-            var isSigned = false;
-
 #if IS_SIGNING_SUPPORTED
-            using (var zip = new ZipArchive(ZipReadStream, ZipArchiveMode.Read, leaveOpen: true))
+            if (!_isSigned.HasValue)
```

Comment:
When debugging `dotnet nuget verify` command, I noticed that `IsSignedAsync` method is invoked multiple times in the same code path. Hence added a nullable bool `_isSigned` to avoid executing the logic multiple times (a minor perf. improvement).

Reply:
nit: Consider positive checks where possible.

Reply:
I am leaning towards keeping the current implementation as is. If you prefer a positive check, I can change the code as shown below. Please let me know your feedback.

``` cs
            if (_isSigned.HasValue)
                return Task.FromResult(_isSigned.Value);

            _isSigned = false;

            using (var zip = new ZipArchive(ZipReadStream, ZipArchiveMode.Read, leaveOpen: true))
            {
                var signatureEntry = zip.GetEntry(SigningSpecifications.SignaturePath);

                if (signatureEntry != null &&
                   string.Equals(signatureEntry.Name, SigningSpecifications.SignaturePath, StringComparison.Ordinal))
                {
                    _isSigned = true;
                }
            }

            return Task.FromResult(_isSigned.Value);
```